### PR TITLE
✨ Expose agent command line flags from controller

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
## Summary
This PR adds command line flags to the controller that expose the functionality of the agent's command line flags about logging.

## Related issue(s)

This embodies the simple answer to #57

### CODE REVIEW

File: cmd/ocm-status-addon/main.go [hunk 1]
```diff
@@ -30,19 +33,31 @@ func main() {
...
+	features := featuregate.NewFeatureGate()
+	err := features.Add(map[featuregate.Feature]featuregate.FeatureSpec{
+		logs.ContextualLogging: {
+			Default:    true,
+			PreRelease: featuregate.Alpha}})
+	if err != nil {
+		panic(err)
+	}
+	logConfig := logs.NewLoggingConfiguration()
+	logs.AddFlags(logConfig, pflag.CommandLine)
```
Comment (pdettori) [inline cmd/ocm-status-addon/main.go:46]: I have the same question I posed in #59, why the need for a feature gate for logging? Is this a requirement to use kube contextual logging ?

Comment (MikeSpreitzer) [reply cmd/ocm-status-addon/main.go:46]: Let's discuss #59 in #59.

**Comment by MikeSpreitzer** [comment]
I will add the flags for request qps and burst.

**Comment by MikeSpreitzer** [comment]
The E2E test in CI showed the expected content in the ManifestWork. Here are the args on the agent command line:

```
              - args:
                - agent
                - --hub-kubeconfig=/var/run/hub/kubeconfig
                - --cluster-name=cluster1
                - --addon-namespace=open-cluster-management-agent-addon
                - --logging-format=text
                - --v=0
                - --vmodule=
                - --hub-burst=10
                - --hub-qps=5
                - --local-burst=10
                - --local-qps=5
                - --log-flush-frequency=5s
```

**Comment by MikeSpreitzer** [comment]
Here is the revised controller command line help:

```
mspreitz@mjs13 ocm-status-addon % go run ./cmd/ocm-status-addon controller -h
Start the addon controller

Usage:
  addon controller [flags]

Flags:
      --agent-hub-burst int                  Allowed burst in requests/sec for accessing the hub from the agent (default 10)
      --agent-hub-qps float                  Max average requests/sec for accessing the hub from the agent (default 5)
      --agent-local-burst int                Allowed burst in requests/sec for accessing the local cluster from the agent (default 10)
      --agent-local-qps float                Max average requests/sec for accessing the local cluster from the agent (default 5)
      --agent-log-flush-frequency duration   Maximum number of seconds between log flushes on the agent (default 5s)
      --agent-logging-format string          Sets the log format. Permitted formats: "text". on the agent (default "text")
      --agent-v Level                        number for the log level verbosity on the agent
      --agent-vmodule pattern=N,...          comma-separated list of pattern=N settings for file-filtered logging (only works for text log format) on the agent
      --component-namespace string           Namespace of the component.
      --enable-leader-election               Enables the leader election for the controller
  -h, --help                                 help for controller
      --kubeconfig string                    Location of the master configuration file to run from.

Global Flags:
      --log-flush-frequency duration   Maximum number of seconds between log flushes (default 5s)
      --logging-format string          Sets the log format. Permitted formats: "text". (default "text")
  -v, --v Level                        number for the log level verbosity
      --vmodule pattern=N,...          comma-separated list of pattern=N settings for file-filtered logging (only works for text log format)
```

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -72,4 +72,5 @@ spec:
           value: STATUS_ADDDON_IMAGE_NAME_PLACEHOLDER
         args:
         - "controller"
+        - "--agent-v=2"
```
Comment (MikeSpreitzer) [inline config/manager/manager.yaml:75]: BTW, I expect this to get overwritten soon as we plumb control over logging through higher levels.

**Comment by ezrasilvera** [comment]
@MikeSpreitzer Looks good - maybe you want to move the default value to the beginning of the line instead the end (I agree the end is more reasonable in terms of structure but for long lines it's a little hidden )

**Comment by MikeSpreitzer** [comment]
The parenthetic remark about the default value is not under programmer control, it is added by the flagging package.

**Comment by MikeSpreitzer** [comment]
@ezrasilvera: The parenthetic remark about the default value is not under programmer control, it is added by the flagging package.

**Review by ezrasilvera — APPROVED** [review]

**Pull request merged**